packages/dev/core/src/Misc/timingTools.ts

```diff
@@ -18,14 +18,14 @@ export class TimingTools {
     }
 }
 
-function _runWithCondition(condition: () => boolean, onSuccess: () => void, onError?: (e?: any) => void) {
+function _runWithCondition(condition: () => boolean, onSuccess: () => void, onError?: (e?: any, wasUnexpected?: boolean) => void) {

```

Comment:
What error would be expected ?

Reply:
I guess it would only be the timeout and I wonder if something else should be done here instead ? cc @RaananW

Reply:
Yes, `onError` is called either when there's a timeout or inside that `catch` in `_runWithCondition`.

Reply:
let s see if we have a better option and if not, let s do the opposite, flag as true by default and force to false only in timeout ?

Reply:
Done.

Reply:
The other option is to use a TimeoutError so we do not grow the list of params and only relies on the error type. Let s see what @RaananW prefers tomorrow.